### ISSUE TITLE
Fix the CleanupExpiredArtifactsFromProxyRepositoriesCronJobTestIT

### ISSUE THREAD

**Issue by carlspring** [strongbox/strongbox#1745]
# Task Description

We need to fix this test against the `issues/1649` branch.

# Stacktrace

```
Error
Lambda expression in org.awaitility.core.ConditionFactory that uses java.util.concurrent.atomic.AtomicBoolean: expected (<true> or <true>) but was <false> within 30 seconds.
Stacktrace
org.awaitility.core.ConditionTimeoutException: Lambda expression in org.awaitility.core.ConditionFactory that uses java.util.concurrent.atomic.AtomicBoolean: expected (<true> or <true>) but was <false> within 30 seconds.
	at org.carlspring.strongbox.cron.jobs.CleanupExpiredArtifactsFromProxyRepositoriesCronJobTestIT.expiredArtifactsCleanupCronJobShouldCleanupDatabaseAndStorage(CleanupExpiredArtifactsFromProxyRepositoriesCronJobTestIT.java:143)
```

# Task Relationships

This task:
* Is a sub-task of: #1649

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @sbespalov
  * @carlspring
  * @steve-todorov

**Comment by valiobojinov** [comment]
I belive I was able to reproduce it here's the stacktrance for convenience
[`12:22:52.386] | E | 1-worker-3 | o.j.g.database.StandardJanusGraph     | Could not commit transaction [26] due to exception
org.janusgraph.diskstorage.locking.PermanentLockingException: Local lock contention
	at org.janusgraph.diskstorage.locking.AbstractLocker.writeLock(AbstractLocker.java:327)
	at org.janusgraph.diskstorage.locking.consistentkey.ExpectedValueCheckingStore.acquireLock(ExpectedValueCheckingStore.java:103)
	at org.janusgraph.diskstorage.keycolumnvalue.KCVSProxy.acquireLock(KCVSProxy.java:51)
	at org.janusgraph.diskstorage.BackendTransaction.acquireEdgeLock(BackendTransaction.java:236)
	at org.janusgraph.graphdb.database.StandardJanusGraph.prepareCommit(StandardJanusGraph.java:551)
	at org.janusgraph.graphdb.database.StandardJanusGraph.commit(StandardJanusGraph.java:731)
	at org.janusgraph.graphdb.transaction.StandardJanusGraphTx.commit(StandardJanusGraphTx.java:1425)
	at org.janusgraph.graphdb.tinkerpop.JanusGraphBlueprintsTransaction$1.doCommit(JanusGraphBlueprintsTransaction.java:185)
	at org.apache.tinkerpop.gremlin.structure.util.AbstractTransaction.commit(AbstractTransaction.java:104)
	at org.opencypher.gremlin.neo4j.ogm.transaction.GremlinTransaction.doCommit(GremlinTransaction.java:111)
	at org.opencypher.gremlin.neo4j.ogm.transaction.GremlinTransaction.commit(GremlinTransaction.java:82)
	at org.springframework.data.neo4j.transaction.Neo4jTransactionManager.doCommit(Neo4jTransactionManager.java:288)
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.processCommit(AbstractPlatformTransactionManager.java:744)
	at org.springframework.transaction.support.AbstractPlatformTransactionManager.commit(AbstractPlatformTransactionManager.java:712)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.commitTransactionAfterReturning(TransactionAspectSupport.java:631)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:385)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:99)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:747)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:88)
	at org.carlspring.strongbox.data.tx.TransactionRetryAspect.proceedWithRetry(TransactionRetryAspect.java:95)
	at org.carlspring.strongbox.data.tx.TransactionRetryAspect.proceed(TransactionRetryAspect.java:79)
	at org.carlspring.strongbox.data.tx.TransactionRetryAspect.transactionalJta(TransactionRetryAspect.java:63)
	at sun.reflect.GeneratedMethodAccessor95.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:644)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:633)
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:70)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:747)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:95)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:747)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:689)
	at org.carlspring.strongbox.repositories.ArtifactRepository$$EnhancerBySpringCGLIB$$41fc160b.delete(<generated>)
	at org.carlspring.strongbox.providers.layout.LayoutFileSystemProvider.doDeletePath(LayoutFileSystemProvider.java:285)
	at org.carlspring.strongbox.providers.io.StorageFileSystemProvider.doDeletePath(StorageFileSystemProvider.java:275)
	at org.carlspring.strongbox.providers.io.StorageFileSystemProvider$3.visitFile(StorageFileSystemProvider.java:236)
	at org.carlspring.strongbox.providers.io.StorageFileSystemProvider$3.visitFile(StorageFileSystemProvider.java:229)
	at java.nio.file.Files.walkFileTree(Files.java:2670)
	at java.nio.file.Files.walkFileTree(Files.java:2742)
	at org.carlspring.strongbox.providers.io.StorageFileSystemProvider.recursiveDeleteExceptRoot(StorageFileSystemProvider.java:228)
	at org.carlspring.strongbox.providers.io.StorageFileSystemProvider.delete(StorageFileSystemProvider.java:206)
	at org.carlspring.strongbox.providers.layout.LayoutFileSystemProvider.delete(LayoutFileSystemProvider.java:262)
	at org.carlspring.strongbox.providers.layout.Maven2FileSystemProvider.delete(Maven2FileSystemProvider.java:80)
	at org.carlspring.strongbox.providers.io.StorageFileSystemProvider.delete(StorageFileSystemProvider.java:178)
	at java.nio.file.Files.delete(Files.java:1126)
	at org.carlspring.strongbox.repository.AbstractRepositoryManagementStrategy.removeDirectoryStructure(AbstractRepositoryManagementStrategy.java:99)
	at org.carlspring.strongbox.repository.AbstractRepositoryManagementStrategy.removeRepository(AbstractRepositoryManagementStrategy.java:85)
	at org.carlspring.strongbox.services.impl.RepositoryManagementServiceImpl.removeRepository(RepositoryManagementServiceImpl.java:103)
	at org.carlspring.strongbox.testing.storage.repository.TestRepositoryContext.closeInternal(TestRepositoryContext.java:268)
	at org.carlspring.strongbox.testing.storage.repository.TestRepositoryContext.close(TestRepositoryContext.java:239)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleElement.invoke(InitDestroyAnnotationBeanPostProcessor.java:389)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMetadata.invokeDestroyMethods(InitDestroyAnnotationBeanPostProcessor.java:347)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeDestruction(InitDestroyAnnotationBeanPostProcessor.java:177)
	at org.springframework.beans.factory.support.DisposableBeanAdapter.destroy(DisposableBeanAdapter.java:242)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroyBean(DefaultSingletonBeanRegistry.java:571)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroySingleton(DefaultSingletonBeanRegistry.java:543)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.destroySingleton(DefaultListableBeanFactory.java:1072)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroySingletons(DefaultSingletonBeanRegistry.java:504)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.destroySingletons(DefaultListableBeanFactory.java:1065)
	at org.springframework.context.support.AbstractApplicationContext.destroyBeans(AbstractApplicationContext.java:1060)
	at org.springframework.context.support.AbstractApplicationContext.doClose(AbstractApplicationContext.java:1029)
	at org.springframework.context.support.AbstractApplicationContext.close(AbstractApplicationContext.java:978)
	at org.carlspring.strongbox.testing.storage.repository.TestRepositoryManagementApplicationContext.close(TestRepositoryManagementApplicationContext.java:274)
	at org.carlspring.strongbox.testing.storage.repository.TestRepositoryManagementApplicationContext.closeExtension(TestRepositoryManagementApplicationContext.java:103)
	at org.carlspring.strongbox.testing.storage.repository.TestRepositoryManagementContextSupport.afterTestExecution(TestRepositoryManagementContextSupport.java:44)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeAfterTestExecutionCallbacks$8(TestMethodTestDescriptor.java:220)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeAllAfterMethodsOrCallbacks$14(TestMethodTestDescriptor.java:245)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeAllAfterMethodsOrCallbacks(TestMethodTestDescriptor.java:243)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeAfterTestExecutionCallbacks(TestMethodTestDescriptor.java:219)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:129)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:135)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:171)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:115)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:171)
	at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
12:22:52.392 | W | 1-worker-3 | o.c.s.data.tx.TransactionRetryAspect  | Retry [0]-[void org.carlspring.strongbox.gremlin.repositories.GremlinRepository.delete(DomainObject)].
12:22:52.497 | E | 1-worker-3 | o.c.s.t.s.r.TestRepositoryContext     | Failed to close [TestRepository] with id [storage-common-proxies/maven-central-dead] 
java.nio.file.NoSuchFileException: /home/valio/git/strongbox/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/target/strongbox-vault/storages/storage-common-proxies/maven-central-dead/org/carlspring/maven/maven-commons/1.2/maven-commons-1.2.jar.sha1
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
	at sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
	at sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:144)
	at sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99)
	at org.carlspring.strongbox.providers.io.StorageFileSystemProvider.readAttributes(StorageFileSystemProvider.java:513)
	at java.nio.file.Files.readAttributes(Files.java:1737)
	at java.nio.file.FileTreeWalker.getAttributes(FileTreeWalker.java:219)
	at java.nio.file.FileTreeWalker.visit(FileTreeWalker.java:276)
	at java.nio.file.FileTreeWalker.next(FileTreeWalker.java:372)
	at java.nio.file.Files.walkFileTree(Files.java:2706)
	at java.nio.file.Files.walkFileTree(Files.java:2742)
	at org.carlspring.strongbox.providers.io.StorageFileSystemProvider.recursiveDeleteExceptRoot(StorageFileSystemProvider.java:228)
	at org.carlspring.strongbox.providers.io.StorageFileSystemProvider.delete(StorageFileSystemProvider.java:206)
	at org.carlspring.strongbox.providers.layout.LayoutFileSystemProvider.delete(LayoutFileSystemProvider.java:262)
	at org.carlspring.strongbox.providers.layout.Maven2FileSystemProvider.delete(Maven2FileSystemProvider.java:80)
	at org.carlspring.strongbox.providers.io.StorageFileSystemProvider.delete(StorageFileSystemProvider.java:178)
	at java.nio.file.Files.delete(Files.java:1126)
	at org.carlspring.strongbox.repository.AbstractRepositoryManagementStrategy.removeDirectoryStructure(AbstractRepositoryManagementStrategy.java:99)
	at org.carlspring.strongbox.repository.AbstractRepositoryManagementStrategy.removeRepository(AbstractRepositoryManagementStrategy.java:85)
	at org.carlspring.strongbox.services.impl.RepositoryManagementServiceImpl.removeRepository(RepositoryManagementServiceImpl.java:103)
	at org.carlspring.strongbox.testing.storage.repository.TestRepositoryContext.closeInternal(TestRepositoryContext.java:268)
	at org.carlspring.strongbox.testing.storage.repository.TestRepositoryContext.close(TestRepositoryContext.java:239)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleElement.invoke(InitDestroyAnnotationBeanPostProcessor.java:389)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMetadata.invokeDestroyMethods(InitDestroyAnnotationBeanPostProcessor.java:347)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeDestruction(InitDestroyAnnotationBeanPostProcessor.java:177)
	at org.springframework.beans.factory.support.DisposableBeanAdapter.destroy(DisposableBeanAdapter.java:242)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroyBean(DefaultSingletonBeanRegistry.java:571)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroySingleton(DefaultSingletonBeanRegistry.java:543)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.destroySingleton(DefaultListableBeanFactory.java:1072)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.destroySingletons(DefaultSingletonBeanRegistry.java:504)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.destroySingletons(DefaultListableBeanFactory.java:1065)
	at org.springframework.context.support.AbstractApplicationContext.destroyBeans(AbstractApplicationContext.java:1060)
	at org.springframework.context.support.AbstractApplicationContext.doClose(AbstractApplicationContext.java:1029)
	at org.springframework.context.support.AbstractApplicationContext.close(AbstractApplicationContext.java:978)
	at org.carlspring.strongbox.testing.storage.repository.TestRepositoryManagementApplicationContext.close(TestRepositoryManagementApplicationContext.java:274)
	at org.carlspring.strongbox.testing.storage.repository.TestRepositoryManagementApplicationContext.closeExtension(TestRepositoryManagementApplicationContext.java:103)
	at org.carlspring.strongbox.testing.storage.repository.TestRepositoryManagementContextSupport.afterTestExecution(TestRepositoryManagementContextSupport.java:44)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeAfterTestExecutionCallbacks$8(TestMethodTestDescriptor.java:220)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeAllAfterMethodsOrCallbacks$14(TestMethodTestDescriptor.java:245)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeAllAfterMethodsOrCallbacks(TestMethodTestDescriptor.java:243)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeAfterTestExecutionCallbacks(TestMethodTestDescriptor.java:219)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:129)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:135)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:171)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:115)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:171)
	at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 25.809 s <<< FAILURE! - in org.carlspring.strongbox.providers.repository.WhenRepositoryIsNotAliveDontCleanExpiredArtifactsTestIT
[ERROR] expiredArtifactsCleanerShouldNotCleanupDatabaseAndStorageWhenRepositoryIsNotAlive{Repository}  Time elapsed: 4.677 s  <<< ERROR!
org.springframework.context.ApplicationContextException: Failed to close context.; nested exception is java.io.IOException: Failed to close following repositories: [maven-central-dead]
	at org.carlspring.strongbox.testing.storage.repository.TestRepositoryManagementApplicationContext.close(TestRepositoryManagementApplicationContext.java:279)
	at org.carlspring.strongbox.testing.storage.repository.TestRepositoryManagementApplicationContext.closeExtension(TestRepositoryManagementApplicationContext.java:103)
	at org.carlspring.strongbox.testing.storage.repository.TestRepositoryManagementContextSupport.afterTestExecution(TestRepositoryManagementContextSupport.java:44)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeAfterTestExecutionCallbacks$8(TestMethodTestDescriptor.java:220)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeAllAfterMethodsOrCallbacks$14(TestMethodTestDescriptor.java:245)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeAllAfterMethodsOrCallbacks(TestMethodTestDescriptor.java:243)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeAfterTestExecutionCallbacks(TestMethodTestDescriptor.java:219)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:129)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:135)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:171)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:115)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:171)
	at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.io.IOException: Failed to close following repositories: [maven-central-dead]
	at org.carlspring.strongbox.testing.storage.repository.TestRepositoryManagementApplicationContext.lambda$repositoriesShouldBeClosed$8(TestRepositoryManagementApplicationContext.java:299)
	at java.util.Optional.map(Optional.java:215)
	at org.carlspring.strongbox.testing.storage.repository.TestRepositoryManagementApplicationContext.repositoriesShouldBeClosed(TestRepositoryManagementApplicationContext.java:298)
	at org.carlspring.strongbox.testing.storage.repository.TestRepositoryManagementApplicationContext.close(TestRepositoryManagementApplicationContext.java:275)
	... 34 more
`

**Comment by valiobojinov** [comment]
[stacktrace.txt](https://github.com/strongbox/strongbox/files/4550867/stacktrace.txt)

**Comment by carlspring** [comment]
This has either already been solved by previous changes, or will be fixed with #1764 .

**Comment by sbespalov** [comment]
most likely it could be fixed same way as #1831

**Comment by steve-todorov** [comment]
This test seems to have been fixed in `issues/1649` so the issue is obsolete. Closing it. Thanks to all participants! :)
